### PR TITLE
refactor: update actionButton to better reusability and add verify to…

### PR DESCRIPTION
**Pull request by luizmuraro**
Description: 
- Refactor Action button reusability on all strategy components
- Add top 100 verification UI

### CODE REVIEW

File: apps/govquests-api/rails_app/db/seeds.rb [hunk 1]
```diff
@@ -70,9 +70,23 @@ def create_action(action_data)
...
+
+verify_position_action = {
+  action_type: "verify_position",
+  display_data: {
+    title: "Verify your position",
+    description: "Click to verify your status as a Top 100 Delegate."
+  },
+  action_data: {
+    action_type: "verify_position",
+    min_position: 100
```
Comment (ribeirojose) [inline apps/govquests-api/rails_app/db/seeds.rb:82]: this wasn't too clear to me. you could perhaps rename it to sth like `should_be_higher_than`

File: apps/govquests-api/rails_app/db/seeds.rb [hunk 2]
```diff
@@ -105,6 +119,18 @@ def create_action(action_data)
...
+  {
+    display_data: {
+      title: "Top 100 Delegates",
+      intro: "Have you reached the Top 100 Delegates in Season 6? Let’s check your ranking and, if you’re among the top, celebrate your achievement with a reward!",
+      image_url: "https://example.com/advanced-governance.jpg",
+      requirements: 'You need to be a delegate to do this quest! If you’re not one, start with Become Delegate Quest.'
+    },
+    quest_type: "Governance",
+    audience: "Delegates",
+    rewards: [{type: "Points", amount: 100}],
```
Comment (ribeirojose) [inline apps/govquests-api/rails_app/db/seeds.rb:133]: 👍 I refactored the seed to be a bit more readable and already merged that code. lmk if you need any help refactoring yours to that format

File: apps/govquests-api/rails_app/db/seeds.rb (hunk 1 above)
Comment (luizmuraro) [reply apps/govquests-api/rails_app/db/seeds.rb:82]: You're right, I was not happy with the name too, I'll change it!

File: apps/govquests-api/rails_app/db/seeds.rb (hunk 2 above)
Comment (luizmuraro) [reply apps/govquests-api/rails_app/db/seeds.rb:133]: @ribeirojose Nice refactor on seed, I had no problem to refactor to that format!

File: apps/govquests-frontend/src/domains/action_tracking/components/ActionButton.tsx [hunk 3]
```diff
@@ -1,52 +1,94 @@
...
+  gitcoin_score: {
+    statuses: Record<GitcoinScoreStatus, StatusConfig>;
+  };
+  read_document: {
+    statuses: Record<ReadDocumentStatus, StatusConfig>;
+  };
+  verify_position: {
+    statuses: Record<VerifyPositionStatus, StatusConfig>;
+  };
+  ens: {
```
Comment (ribeirojose) [inline apps/govquests-frontend/src/domains/action_tracking/components/ActionButton.tsx:28]: 💙

File: apps/govquests-frontend/src/domains/action_tracking/strategies/EnsStrategy.tsx
Comment (ribeirojose) [inline apps/govquests-frontend/src/domains/action_tracking/strategies/EnsStrategy.tsx:1]: ty for cleaning this up 😬

File: apps/govquests-frontend/src/domains/action_tracking/types/actionButtonTypes.ts [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+
+export type VerifyPositionStatus = "unstarted" | "started" | "completed";
+
+export type EnsStatus = "unstarted" | "started" | "completed";
+
+export type ActionType =
+  | "gitcoin_score"
+  | "read_document"
+  | "verify_position"
+  | "ens";
```
Comment (ribeirojose) [inline apps/govquests-frontend/src/domains/action_tracking/types/actionButtonTypes.ts:17]: we should probably create enums in the gql API to make this typing easier 🤔

File: apps/govquests-frontend/src/domains/action_tracking/components/ActionButton.tsx [hunk 5]
```diff
@@ -1,52 +1,94 @@
...
+      started: { label: "Verifying ENS", icon: null },
+      completed: { label: "ENS Connected", icon: null },
+    },
   },
 };
 
-const ActionButton: React.FC<ActionButtonProps> = ({
+type ActionButtonProps<T extends ActionType> = {
+  actionType: T;
+  status: keyof ActionConfig[T]["statuses"];
```
Comment (ribeirojose) [inline apps/govquests-frontend/src/domains/action_tracking/components/ActionButton.tsx:76]: typescript 🪄

**Review by ribeirojose — APPROVED** [review]
looking good! I'm still wondering about the scalability of our approach in adding actions. it still seems to me like there's a lot of repeated code and multiple places to change when adding a new action type... I'm wondering how that could be made simpler. has this bothered you so far?

**Comment by luizmuraro** [comment]
> looking good! I'm still wondering about the scalability of our approach in adding actions. it still seems to me like there's a lot of repeated code and multiple places to change when adding a new action type... I'm wondering how that could be made simpler. has this bothered you so far?

Yes, its bothering me too, I think it still needs a lot of refactor, after my current quest I'll try to think in a better solution for all the cases!

**Pull request merged**